ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -409,4 +403,12 @@ boolean empty() {
             return this.resourcesList.size() == 0;
         }
     }
+
+    private static class IngestionResourceSet {
```

Comment:
To be clear - the problem we had before is that we replaced the queues one at a time, and this fixes at by replacing them all at once?

Reply:
Not exactly 
we were setting this.containers for example to a new empty list 
while it was not lock protected and could be read empty at this point and even not reverted back on failure too ...